src/common/io.cc

```diff
@@ -369,7 +369,7 @@ int64_t read_vector(int fd, int64_t *type, std::vector<uint8_t> &buffer) {
   if (closed) {
     goto disconnected;
   }
-  if (length > buffer.size()) {
+  if (static_cast<size_t>(length) > buffer.size()) {
```

Comment:
Why not just have `length` be `uint64_t`  (I'm assuming `size_t` is not an option)?  
It's better to let the data type match the semantics of the variable than to check for it. Otherwise if you pass around signed variables for unsigned quantities then you'll keep having to worry about whether a negative value is possible and if it might have a special meaning.